api/metrics/taxitrips.py

```diff
@@ -0,0 +1,35 @@
+from api.utils.database import rows_to_dicts
+
+#multiple metrics for the taxi trip dataset
+class TaxiTripMetrics:
+    def __init__(self, connection):
+        self.connection = connection
+        
+        
+#gets the average speed of a taxi 
+    def avgtaxispeed(self):
+        query = """
+        SELECT
+            trip_miles,
+            trip_minutes/60 as trip_hours,
+            trip_miles/trip_hours as trip_speed
```

Comment:
I think this will throw an error where SQL the column trip_hours is not defined, because it gets defined at the same time as where it is being used. But maybe I am wrong about that.

Reply:
Yeah so it does throw an error because it's defined at the same time as it's being used. I could add a new column to the dataset for trip_hours but would I have to do the transforming and loading steps again?

Reply:
No need to add it as a column in the offline pipeline, there are two ways you can do it in the query:

1. Repeat the expression:

```sql
SELECT
  trip_minutes / 60 as trip_hours,
  trip_miles / (trip_minutes / 60) as avg_speed
FROM taxitrips
...
```

2. In a subquery:

```sql
SELECT
  trip_hours,
  trip_miles / trip_hours as avg_speed
FROM (
  SELECT
    *,
    trip_minutes / 60 as trip_hours
  FROM taxitrips
)
...
```

Approach #1 should work for this case.